Attempt at updated eatlocal for v2 of the platform

rhelmstedter opened the pull request:
Updates:
- used playwright as discussed
- uses the new platform
- completely breaks with the old version
- installs the chromium browser during the initialization no more need for chrome/chromedriver or even for users to run `playwright install`
- updated readme
- no longer pushes code to GitHub when submitting (though I have a function that we can use for a new command if we want)

Still needed:
- tests

rhelmstedter commented:
I think it is good enough to publish. Test coverage is at 80% (which is waaaaay better that it ever was with v1).

bbelderbos commented on src/eatlocal/eatlocal.py:
```diff
@@ -0,0 +1,524 @@
+    url: str = None
+    platform_content: str = None
+
+    def bite_url_to_dir(self, pybites_repo: Path) -> Path:
```
would be more like a property?

rhelmstedter replied:
I have to pass in the pybites repo. Can I do that with a property?

bbelderbos commented on src/eatlocal/eatlocal.py:
```diff
@@ -0,0 +1,524 @@
+    """
+    page: Page = browser.new_page()
+    # only shorten for debugging, some bites need in e2e test need longer
+    page.set_default_timeout(30000)
```
constant?

bbelderbos commented on src/eatlocal/eatlocal.py:
```diff
@@ -0,0 +1,524 @@
+        print("Retrieving bites list...")
+    r = requests.get(EXERCISES_URL)
+    if r.status_code != 200:
+        return
```
should we print something in verbose mode here?

rhelmstedter replied:
I was thinking about getting rid of the verbose mode in general. Replace it with rich Status. That way it shows what's happening but doesn't clog up the terminal.

bbelderbos replied:
Good idea

bbelderbos commented on src/eatlocal/eatlocal.py:
```diff
@@ -0,0 +1,524 @@
+            bites[bite_name] = bite_link
+        except IndexError:
+            continue
+    environ["FZF_DEFAULT_OPTS"] = FZF_DEFAULT_OPTS
```
seems done twice, can we deduplicate and move to the top of the module?

bbelderbos commented on src/eatlocal/eatlocal.py:
```diff
@@ -0,0 +1,524 @@
+            continue
+        if write:
+            bite_description_str += line + "\n"
+        if """end author and learning paths""" in line:
```
cannot remember this, is this something we do in each Bite?

rhelmstedter replied:
It seems like it. It was difficult to parse just the bite description. Because the links to the learning path would show up (but not work) when I tried to display via rich.

bbelderbos replied:
ah I see now, so I better do not remove this html comment then:

```
$ ag "end author"
bites/templates/bite.html
82:            </div><!-- end author and learning paths -->
```

bbelderbos commented on src/eatlocal/eatlocal.py:
```diff
@@ -0,0 +1,524 @@
+    """Create a directory for the bite and write the bite content to it.
+
+    Args:
+        bite: Bite object.
```
seems already covered by type hints

bbelderbos commented on src/eatlocal/eatlocal.py:
```diff
@@ -0,0 +1,524 @@
+            page.click("#validate-button")
+            page.wait_for_selector("#feedback", state="visible")
+            page.wait_for_function(
+                "document.querySelector('#feedback').innerText.includes('test session starts')"
```
this is awesome!

rhelmstedter replied:
Well, you wrote it lol. I copied that from the files you sent me.

bbelderbos commented on src/eatlocal/eatlocal.py:
```diff
@@ -0,0 +1,524 @@
+            )
+
+            validate_result = page.text_content("#feedback")
+    if "Congrats, you passed this Bite" in validate_result:
```
better not touch this then ;)
(I think this code is pretty static at this point)

bbelderbos commented on src/eatlocal/eatlocal.py:
```diff
@@ -0,0 +1,524 @@
+        webbrowser.open(bite.url)
+
+
+def push_to_github(bite: str, bites_repo: str, verbose: bool = False) -> None:
```
do we still want to support this @rhelmstedter? I don't mind dropping github (leaking solutions) if it's no longer needed

rhelmstedter replied:
Yeah, I have no problem taking this out. Just didn't want to delete a bunch of code that was already written if we would use it.

bbelderbos reviewed (approved):
Awesome! I am happy with this so far, the only doubt I have is about pushing to github, can we delete that? 
Apart from that happy to merge, we can always do some follow up work in a new branch.
Thanks Russell!

rhelmstedter commented on tests/test_units.py:
```diff
@@ -1,130 +1,157 @@
+
+    create_bite_dir(bite, testing_config)
+    html_file = bite_dir / "bite.html"
+    python_file = bite_dir / "summing.py"
```
yes

rhelmstedter commented on pyproject.toml:
```diff
@@ -1,33 +1,31 @@
 [build-system]
-requires = ["poetry-core>=1.0.0"]
-build-backend = "poetry.core.masonry.api"
+requires = ["hatchling"]
```
Nope. I have no idea how to do that.

bbelderbos replied:
Nws I will log an issue for that now ...

rhelmstedter commented on src/eatlocal/constants.py:
```diff
@@ -0,0 +1,12 @@
+from pathlib import Path
+
+EATLOCAL_HOME = Path().home() / ".eatlocal"
+BITE_URL = "https://pybitesplatform.com{bite_name}"
```
No, the extra forward slash comes from the bite_name itself. Otherwise there are two.

rhelmstedter commented on src/eatlocal/__main__.py:
```diff
@@ -0,0 +1,173 @@
+    bite.platform_content = download_bite(bite, config, verbose)
+    if bite.platform_content is None:
+        return
+    create_bite_dir(bite, config, verbose, force)
```
Yes

rhelmstedter commented on README.md:
```diff
@@ -111,92 +88,6 @@ pip install eatlocal
-2. Run `eatlocal init` to configure your PyBites username, PyBites password*, and local git repository.
-2. Make sure you have Chrome and chromedriver installed and on `$PATH`. The chromedriver version must match the version of your Chrome browser. [Chromedriver downloads](https://chromedriver.chromium.org/downloads).
-
+Run `eatlocal init` to configure your PyBites username, PyBites password*, and local git repository.
```
eatlocal v1 never supported goolge/github login. We gave users the option to set a password in their profile/dashboard page (I don't remember which one).

bbelderbos replied:
Ah and we did not port that to v2 :)
I will log an issue for this, we can address it later ...

rhelmstedter commented on src/eatlocal/constants.py:
```diff
@@ -0,0 +1,12 @@
+PROFILE_URL = "https://pybitesplatform.com/accounts/profile/"
+FZF_DEFAULT_OPTS = "--height 13 --layout=reverse --border rounded --margin=2%,5%,10%,2%"
+
+WARNING = "red"
```
I need the values to be strings to pass into rich.console. Can you do that with enums?

bbelderbos replied:
`.value` it seems:

```
from enum import Enum

class ConsoleStyle(Enum):
    SUCCESS = "green"
    WARNING = "yellow"
    ERROR = "red"

# Usage with Rich
from rich.console import Console

console = Console()
console.print("This is a success message", style=ConsoleStyle.SUCCESS.value)
console.print("This is a warning message", style=ConsoleStyle.WARNING.value)
console.print("This is an error message", style=ConsoleStyle.ERROR.value)
```

rhelmstedter commented:
@bbelderbos I addressed everything you pointed out with the exception of the two issues you create. I also added some more tests, markers for the slow end 2 end tests, and updated the readme and version to 1.0.0. Going to merge and publish.

bbelderbos commented:
Thanks a lot, let me update PyPI ...
